basic_network_test.go

```diff
@@ -870,43 +870,52 @@ func TestSendTranscodeResponse(t *testing.T) {
 	glog.Infof("\n\nTesting Handle Transcode Result...")
 	//n1 -> n2 -> n3, n3 should get the result, n2 should relay it, n1 should send it.
 	n1, n2 := setupNodes(t, 15000, 15001)
-	n3, n4 := simpleNodes(15003, 15004)
+	n3, _ := simpleNodes(15003, 15004)
 	defer n1.NetworkNode.PeerHost.Close()
 	defer n2.NetworkNode.PeerHost.Close()
 	defer n3.PeerHost.Close()
-	defer n4.PeerHost.Close()
 	connectHosts(n1.NetworkNode.PeerHost, n2.NetworkNode.PeerHost)
 	connectHosts(n2.NetworkNode.PeerHost, n3.PeerHost)
 
 	//Set up n3 to capture the message
 	rc := make(chan map[string]string)
 	n3.PeerHost.SetStreamHandler(Protocol, func(s net.Stream) {
 		defer s.Close()
-		ws := NewBasicStream(s)
-		msg, err := ws.ReceiveMessage()
-		if err != nil {
-			t.Errorf("Error: %v", err)
+		ws := NewBasicInStream(s)
+		for {
+			msg, err := ws.ReceiveMessage()
+			if err != nil {
+				glog.Infof("Error: %v", err)
+				break
+			}
+			glog.Infof("n3 got Msg: %v", msg)
+			rc <- msg.Data.(TranscodeResponseMsg).Result
 		}
-		glog.Infof("n3 got Msg: %v", msg)
-		rc <- msg.Data.(TranscodeResponseMsg).Result
 	})
 
 	strmID := fmt.Sprintf("%vstrmID", peer.IDHexEncode(n3.Identity))
 	//Send the message
-	err := n1.SendTranscodeResponse(peer.IDHexEncode(n3.Identity), strmID, map[string]string{"strmid1": lpms.P240p30fps4x3.Name, "strmid2": lpms.P360p30fps4x3.Name})
-	if err != nil {
-		t.Errorf("Error sending transcode result: %v", err)
-	}
-	select {
-	case r := <-rc:
-		if r["strmid1"] != lpms.P240p30fps4x3.Name {
-			t.Errorf("Expecting %v, got %v", lpms.P240p30fps4x3.Name, r["strmid1"])
+	go func() {
+		for i := 0; i < 3; i++ {
```

Comment:
Is 3 just arbitrary saying "try 3 times"?

Reply:
Yes.